crates/evm/src/evm/tests.rs

```diff
@@ -33,7 +32,7 @@ fn simple_contract_execution_sov_state() {
     simple_contract_execution(evm_db);
 }
 
-fn simple_contract_execution<DB: Database<Error = Infallible> + DatabaseCommit + InitEvmDb>(
+fn simple_contract_execution<DB: Database<Error = DBError> + DatabaseCommit + InitEvmDb>(
```

Comment:
Actually no need to specify `<Error =..>`

Reply:
I think we need to do, it threw this error
```
note: trait defined here, with 0 generic parameters
sov-demo-prover-guest-mock:   --> /Users/erce/.cargo/registry/src/index.crates.io-6f17d22bba15001f/revm-primitives-3.1.0/src/db.rs:11:11
sov-demo-prover-guest-mock:    |
sov-demo-prover-guest-mock: 11 | pub trait Database {
sov-demo-prover-guest-mock:    |           ^^^^^^^^
sov-demo-prover-guest-mock: help: replace the generic bound with the associated type
sov-demo-prover-guest-mock:    |
sov-demo-prover-guest-mock: 78 |     DB: Database<Error = DBError> + DatabaseCommit,
```